models/chainlink/bnb/chainlink_bnb_price_feeds_hourly.sql

```diff
@@ -24,7 +24,7 @@ WITH
    FROM
     {% if not is_incremental() %}
       unnest(
-        sequence(0, CAST(date_diff('hour', timestamp '{{project_start_date}}', current_timestamp) AS INTEGER))
```

Comment:
On the other hand, this is required.

I think we need to do a primer on SQL syntax for literals:
You can specify timestamp _literals_ in two ways:
1. `timestamp '2023-01-01'`
2. cast('2023-01-01' as timestamp)

This syntax is only valid for _literals_. For general timestamp conversions you must use casts. That is, `cast(some_column as timestamp)`.

The rule is:
Use `timestamp '2023-01-01'` for hard coded values (which project_start_date is), use `cast(column_name as timestamp)` for converting columns. You can use casts in both cases, but it's cleaner to use timestamp literals for hard coded values.

Reply:
Yes, this is correct, except for the extra parenthesis in the cast.

It's true that Trino is a bit more picky on types than Postgres or Spark. In Postgres or Spark you are probably used to being able to compare strings directly with date and timestamp types, while this is not possible in Trino. For instance,
```
select block_time > '2023-01-01'
```
is valid in Postgres or Spark, but must be changed to
```
select block_time > timestamp '2023-01-01'
```
in Trino.

Similarly, if you happen to have a column with dates as varchar (though, this should not happen), you would need to do a cast before comparing, so we would have to change
```
select  block_time > some_column_with_varchar_dates
```
to
```
select  block_time > cast(some_column_with_varchar_dates as timestamp)
```

Except for that it should work mostly the same. Between different timestamp and date types Trino does automatic conversions, similar to Postgres and Spark. It's only between varchar types and timestamp types that Trino does not do automatic conversions, while Postgres and Spark do.